src/goto-symex/build_goto_trace.cpp

```diff
@@ -133,6 +133,70 @@ exprt adjust_lhs_object(
 
 /*******************************************************************\
 
+Function: hide_dynamic_object
+
+  Inputs:
+
+ Outputs:
+
+ Purpose: hide variable assignments related to dynamic_object[0-9]
+          and dynamic_[0-9]_array.
+
+\*******************************************************************/
+
+void hide_dynamic_object(const exprt &expr, goto_trace_stept &goto_trace_step)
+{
+  if (expr.id()==ID_symbol)
+  {
+    std::string identifier = expr.get(ID_identifier).c_str();
+    if (identifier.find("dynamic_")!=std::string::npos)
```

Comment:
Yes, I also agree that this looks like a hack. I was thinking about alternatives to get rid of this hack. What do you think if we create a new field called "dynamic_object" that is properly set during the convert method call in goto-programs? This might create some additional overhead when we build the respective goto-program.